pkg/webhook/controlplane/ensurer.go

```diff
@@ -163,62 +157,64 @@ func (e *ensurer) EnsureAdditionalProvisionUnits(ctx context.Context, gctx gcont
         [Install]
         WantedBy=local-fs.target
 `
-	operatingsystems := make(map[string]int)
-	for _, worker := range cluster.Shoot.Spec.Provider.Workers {
-		if worker.DataVolumes != nil {
-			// fail if multiple OSes are used
-			operatingsystems[worker.Machine.Image.Name]++
-			if len(operatingsystems) > 1 {
-				return fmt.Errorf("multiple operatingsystems used: %v; Only one allowed", operatingsystems)
-			}
-			switch worker.Machine.Image.Name {
-			case "flatcar":
-				extensionswebhook.AppendUniqueUnit(new, extensionsv1alpha1.Unit{
-					Name:    "lvm-setup.service",
-					Enable:  ptr.To(true),
-					Command: ptr.To(extensionsv1alpha1.CommandStart),
-					Content: ptr.To(lvmSetup),
-				})
-
-				extensionswebhook.AppendUniqueUnit(new, extensionsv1alpha1.Unit{
-					Name:    "var-lib-containerd.mount",
-					Enable:  ptr.To(true),
-					Command: ptr.To(extensionsv1alpha1.CommandStart),
-					Content: ptr.To(lvmMountContainerd),
-				})
-
-				return nil
-			}
+	volume, err := hasVolume(ctx, gctx)
+	if err != nil {
+		return err
+	}
+	if volume {
```

Comment:
early return saves indentation:
```suggestion
	if !volume {
	   return nil,
	}
```

maybe with a log that we kjnow that we did not detect a volume